docs/topics/classes.md

```diff
@@ -1,245 +1,747 @@
-    Constructors(1)
+    // Creates an object of the 'Person' class using the secondary constructor
+    Person(1)
 }
+//sampleEnd
 ```
 {kotlin-runnable="true"}
 
-If a non-abstract class does not declare any constructors (primary or secondary), it will have a generated primary constructor
-with no arguments. The visibility of the constructor will be public.
+When the `Person(1)` object is created, the initializer block (`init {}`) executes first, followed by the secondary constructor's print statement.
+
+If there's a class that is a [non-abstract class](#abstract-classes) and does not declare any constructors (primary or secondary), it will have a generated primary constructor
+with no arguments:
+
+```kotlin
+// Class with no explicit constructors
+class Person {
+    // No primary or secondary constructors declared
+}
 
+fun main() {
+    // Creates an object of the 'Person' class using the automatically generated primary constructor
+    val person = Person()
+}
+```
+
+The visibility of the constructor will be public, meaning it can be accessed from anywhere.
 If you don't want your class to have a public constructor, declare an empty primary constructor with non-default visibility:
 
 ```kotlin
-class DontCreateMe private constructor() { /*...*/ }
+class Person private constructor() { /*...*/ }
 ```
 
-> On the JVM, if all of the primary constructor parameters have default values, the compiler will generate an additional parameterless constructor which will use the default values. This makes it easier to use Kotlin with libraries such as Jackson or JPA that create class instances through parameterless constructors.
+> On the JVM, if all primary constructor parameters have default values, the compiler generates 
+> an additional parameterless constructor that uses those default values. 
+> 
+> This makes it easier to use Kotlin with libraries such as [Jackson](https://github.com/FasterXML/jackson) 
+> or [Spring Data JPA](https://spring.io/projects/spring-data-jpa), which create class objects through 
+> parameterless constructors.
 >
+> In the following example, Kotlin automatically generates a parameterless constructor `Person()` that uses the default value `""`:
+> 
 > ```kotlin
-> class Customer(val customerName: String = "")
+> class Person(val personName: String = "")
 > ```
 >
 {style="note"}
 
-## Creating instances of classes
+## Creating objects of classes
 
-To create an instance of a class, call the constructor as if it were a regular function. You can assign the created instance to a [variable](basic-syntax.md#variables):
+Like in other object-oriented programming languages, a class in Kotlin serves as a blueprint or template 
+that defines the properties (attributes) and behaviors (functions) of objects. When you create an instance 
+of a class, you are creating a concrete object based on that blueprint.
+
+To create an object of a class, use the class name followed by parentheses (`()`), similar to calling a function:
```

Comment:
In the previous paragraphs we were already creating objects of classes using this syntax. Maybe we should briefly introduce how to call a constructor before, and here we are explaining that in more detail?

Reply:
Added it in primary constructor, right before the first runnable example of object creation.